### PR TITLE
Add instances for MySQL JSON as Aeson Value.

### PR DESCRIPTION
**Pull request by skeet70**
Adds support for [MySQL JSON](https://dev.mysql.com/doc/refman/8.0/en/json.html) as an Aeson Value.

### CODE REVIEW

File: Database/Beam/MySQL/Syntax.hs [hunk 1]
```diff
@@ -508,6 +510,9 @@ instance HasSqlValueSyntax MysqlValueSyntax x => HasSqlValueSyntax MysqlValueSyn
     sqlValueSyntax Nothing = sqlValueSyntax SqlNull
     sqlValueSyntax (Just x) = sqlValueSyntax x
 
+instance HasSqlValueSyntax MysqlValueSyntax A.Value where
+    sqlValueSyntax = MysqlValueSyntax . escape . BL.toStrict . A.encode
```
Comment (skeet70) [inline Database/Beam/MySQL/Syntax.hs]: This is wrong. It produces the right output but isn't escaped/quoted correctly.

`MysqlValueSyntax . (\x -> emit "\"" <> x <> emit "\"") . escape . toStrict . encode` seems to work correctly, but that may be too much. I don't know if the `escape` is necessary here? Not sure what the security implications of unescaped SQL inside of JSON objects sent into the DB are

Comment (skeet70) [reply Database/Beam/MySQL/Syntax.hs]: `"INSERT INTO `saved_blueprints`(`saved_blueprint_id`, `user_id`, `name`, `currency`, `default_blueprint`, `charts`)VALUES (DEFAULT, 40240, 'WHAAAAT', 'JPY', FALSE, [1,2,3])"`

vs 

`"INSERT INTO `saved_blueprints`(`saved_blueprint_id`, `user_id`, `name`, `currency`, `default_blueprint`, `charts`)VALUES (DEFAULT, 40240, 'WHAAAAT', 'JPY', FALSE, \"[1,2,3]\")"` for comparison.

Comment (skeet70) [reply Database/Beam/MySQL/Syntax.hs]: For the input:

```json
{"mascot": "Our mascot is a dolphin named \"Sakila\"."}
```
,

```haskell
instance HasSqlValueSyntax MysqlValueSyntax Value where
  sqlValueSyntax v =
    MysqlValueSyntax $
    MysqlSyntax
      (\next doEscape before conn -> do
         escaped <- doEscape (toStrict $ encode v)
         next doEscape (before <> "'" <> byteString escaped <> "'") conn)
    -- MysqlValueSyntax .
    -- (\x -> emit "'" <> x <> emit "'") . escape . toStrict . encode
```
(trying to emulate how text does it, but it's the same as my commented attempt)
produces

```mysql
'{\\\"mascot\\\":\\\"Our mascot is a dolphin named \\\\\\\"Sakila\\\\\\\".\\\"}'
```

Which does work correctly in the DB, but according to [these docs]() should be:

```mysql
'{"mascot": "Our mascot is a dolphin named \\"Sakila\\"."}'
```

**Comment by skeet70** [comment]
@tathougies it looks like [here](https://github.com/tathougies/beam/blob/5fb5f168b7c36516082da667af75ccfff9a83bc2/beam-postgres/Database/Beam/Postgres/PgSpecific.hs#L413-L425) it ends up functioning pretty similarly? This is directly using `Value` as the type instead of `PgJson a` because this is to support actually using JSON Values, not your own values stored as JSON in the DB. The escaping is working.

**Comment by tathougies** [comment]
I see... Great work!

**Pull request merged**